indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.overlord.supervisor;
+
+import com.google.common.base.Optional;
+import com.metamx.common.lifecycle.LifecycleStart;
+import com.metamx.common.lifecycle.LifecycleStop;
+import com.metamx.emitter.EmittingLogger;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Manages the creation and lifetime of {@link Supervisor}.
+ */
+public class SupervisorManager
+{
+  private static final EmittingLogger log = new EmittingLogger(SupervisorManager.class);
+
+  private final Map<String, Supervisor> supervisors = new HashMap<>();
+
+  public Map<String, Supervisor> getSupervisors()
+  {
+    return supervisors;
+  }
+
+  public Optional<Supervisor> getSupervisor(String id)
```

Comment:
It looks like all the callers of this are actually just wanting the spec, so it should simplify things to do:

- Change this to `getSupervisorSpec`
- Remove `getSpec` from Supervisor (I don't think anything would need that anymore)